lib/lists.nix

```
@@ -612,6 +616,42 @@ rec {
       if len < 2 then list
       else (sort strictLess pivot.left) ++  [ first ] ++  (sort strictLess pivot.right));
 
+  /*
+    Sort a list based on the default comparison of a derived property `b`.
+
+    This calls the function only once per element, so if your function is non-trivial,
+    it is easily twice as efficient as an unprepared [`sort`](#function-library-lib.lists.sort), and then a log(n) factor.
+
+    However, if the passed function is very cheap, `sort` may be cheaper,
+    because `sort` does not allocate an intermediate list of `(a, b)` pairs.
```

A bit too implementation-detail heavy imo.

```suggestion
    However, if the passed function is very cheap, `sort` may be cheaper.
```